render/render.js

```diff
@@ -775,14 +775,19 @@ module.exports = function($window) {
 		) && key in vnode.dom
 	}
 
+	var matchUpperCase = /[A-Z]/g
+	function prependDashAndLowerCase(string){
+		return "-" + string.toLowerCase()
+	}
+
 	//style
 	function updateStyle(element, old, input) {
 		if(typeof input === "object") {
 			var style = {}
 			for(var key in input) {
-				style[key] = String(input[key]).replace(/[A-Z]/g, function(capital){
-					return "-" + capital.toLowerCase()
-				})
+				style[key] = key[0] === "-" && key[1] === "-"
```

Comment:
@porsager Wait a second...I think you got your key/value manipulation flipped. This is what you need to be doing to the *key*, not the *value*.

Reply:
Lol.. that was actually there from before, I just didn't spot it either 🤣

I think there's something fundamentally wrong here.. why are we kebab-casing the style properties?

Reply:
@porsager It's per #2308 + some Gitter discussion about a week ago, where I suggested that as a stopgap solution so we could move to `setProperty`/etc.

Reply:
Ah, ok to use setProperty instead.. So the tests needs to be fixed and the key/value mess..

Can I just do it in this PR ?

Reply:
Sure, although I already have a partially-written patch laying around for the cleanup.

<details>
<summary>Here's the code I have so far, untested</summary>

This one grows the bundle a little bit, but it's like 16 bytes.

```js
var uppercaseRegex = /[A-Z]/g
function toLowerCase(capital) { return "-" + capital.toLowerCase() }
function normalizeKey(key) {
	return key[0] === "-" && key[1] === "-" ? key :
		key === "cssFloat" ? "float" :
			key.replace(uppercaseRegex, toLowerCase)
}
function updateStyle(element, old, style) {
	if (old === style) {
		// Do nothing.
	} else if (style == null || typeof style !== "object") {
		element.style.cssText = ""
	} else if (old == null || typeof old !== "object") {
		if (typeof old !== "object") element.style.cssText = ""
		// `old` is missing or a string, `style` is an object.
		// Add new style properties
		for (var key in style) {
			var value = style[key]
			if (value != null) element.style.setProperty(normalizeKey(key), String(value))
		}
	} else {
		// Both old & new are (different) objects.
		// Update style properties that have changed
		for (var key in style) {
			var value = style[key]
			if (value != null && (value = String(value)) !== String(old[key])) {
				element.style.setProperty(normalizeKey(key), value)
			}
		}
		// Remove style properties that no longer exist
		for (var key in old) {
			if (old[key] != null && style[key] == null) {
				element.style.removeProperty(normalizeKey(key))
			}
		}
	}
}
```
</details>

Reply:
I haven't pushed it anywhere, though, and that was like 5-10 minutes of screwing around while watching this thread.

Reply:
ah ok, well you decide ;) I remember looking/talking about `updateStyle` about a half year ago, and there was a few edge cases that didn't seem straight forward to me, so perhaps we should start by making this PR work, and improve afterwards?

Reply:
We can defer it to a follow-up PR. BTW, the edge cases are pretty straightforward:

- If the new style is missing, we just clear it.
- If the old style is missing, we just add the new style.
- If neither the old or new styles are objects, we can just replace the `cssText` and let the browser do all the work. It has to diff anyways.
- If the old style is an object and the new style is not, the browser diffs it for us if we use `cssText`.
- If the old style is not an object and the new style is, it's easier to clear and start over than it is to parse the style string and diff it ourselves.
- If both the old and new styles are objects, we have to diff and patch it directly.

Of course, there's a lot of duplicate steps there, but that's my basic line of thinking and why I came up with what I put in that comment.

Reply:
Sooorrry 😅

Reply:
Good luck with that @porsager, from my experience it's not exactly a quick fix

Reply:
Yeah, that's what I experienced the last time I dove into it, but I'm ready for another go 😄

Reply:
@barneycarroll It turned out to be [a quick fix](https://github.com/MithrilJS/mithril.js/pull/2311/commits/6d711ee5cd3abb1d11b5c5cf3d35c58c2fc9ff6d).

Reply:
Oh.. I thought @barneycarroll meant refactoring `updateStyle` 😛

Reply:
@porsager No, @barneycarroll had a seriously difficult time (nicely detailed on Gitter about a week or so ago) fixing the tests and DOM mocks especially to work with his `setProperty` update, and I had to advise dropping a quick hack to make the few hundred failing tests pass.

Reply:
Perhaps he wasn't ready to add `return true` at the top of the equals function in ospec as I did...